include/model_checker/statistical_model_checking_engine.hpp

```diff
@@ -55,22 +55,56 @@ namespace model_checker {
 template<typename StateType, typename ValueType>
 class StateGeneration;
 
+/*!
+ * @brief The implementation of the ModelChecking engine
+ * @tparam ModelType Definition of the kind of model to evaluate (e.g. DTMC, MDP, ...)
+ * @tparam StateType Variable type used to identify the states in the model
+ */
 template<typename ModelType, typename StateType = uint32_t>
-class StatisticalExplorationModelChecker : public storm::modelchecker::AbstractModelChecker<ModelType> {
+class StatisticalModelCheckingEngine : public storm::modelchecker::AbstractModelChecker<ModelType> {
    public:
     typedef typename ModelType::ValueType ValueType;
     typedef StateType ActionType;
 
-    StatisticalExplorationModelChecker(storm::storage::SymbolicModelDescription const& in_model, const settings::SmcSettings& settings);
+    /*!
+     * @brief Constructor for the StatisticalModelCheckingEngine
+     * @param in_model the model to perform the evaluation on
+     * @param settings A collection of settings, used to configure the engine
+     */
+    StatisticalModelCheckingEngine(storm::storage::SymbolicModelDescription const& in_model, const settings::SmcSettings& settings);
 
+    /*!
+     * @brief Check if the provided property is supported by the engine
+     * @param check_task The property to verify
+     * @return true if the property can be handled, false otherwise
+     */
     virtual bool canHandle(storm::modelchecker::CheckTask<storm::logic::Formula, ValueType> const& check_task) const override;
 
+    /*!
+     * @brief Static version of the engine support check for the provided property
```

Comment:
supports?

Reply:
It was a "support check", guess can be changed to compatibility check